File: Linked_List/019.Remove-Nth-Node-From-End-of-List/019.Remove-Nth-Node-From-End-of-List.ts
function removeNthFromEnd(head: ListNode | null, n: number): ListNode | null {
  const dummy = new ListNode(0, head);
  let fast = head;
  let slow = head;
  let prev = dummy;
  while (n > 0) {
    fast = fast.next;
    n--;
  }

  while (fast !== null) {
    prev = slow;
    fast = fast.next;
    slow = slow.next;
  }

  prev.next = slow.next;

  return dummy.next;
}

File: Linked_List/019.Remove-Nth-Node-From-End-of-List/019.Remove-Nth-Node-From-End-of-List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        dummy = ListNode(0, head)
        fast = dummy
        slow = dummy
        
        while n > 0:
            fast = fast.next
            n -= 1

        while fast and fast.next:
            fast = fast.next
            slow = slow.next
        print(slow.val)
        slow.next = slow.next.next

        return dummy.next
